tests/util/test_async_helpers.py

```diff
@@ -72,8 +75,8 @@ def check_val(res, idx):
         self.assertEqual(results[0], 123, "observer 1 callback result")
         self.assertEqual(results[1], 123, "observer 2 callback result")
 
-    def test_failure(self):
-        origin_d = Deferred()
+    def test_failure(self) -> None:
+        origin_d: Deferred = Deferred()
```

Comment:
I left it as a bare `Deferred` because it never has a value associated (and the value doesn't matter). I can change it to `Deferred[None]`, but that also sounds like lying?

Reply:
(Let me know if you want a change, but I'm tempted to leave it.)

Reply:
Ahh I see. `Deferred[NoReturn]` might work as a "this deferred will never succeed"? But I think we're splitting hairs and it's not worth it for test code. Let's leave it be.

Reply:
Oh, perhaps? I have no idea if that'll work.

I agree, don't think it is worth it for tests.